app/views/activity_notification/subscriptions/default/_subscription.html.erb

```diff
@@ -24,12 +24,12 @@
           <% end %>
         <% else %>
           <% if ActivityNotification.config.subscribe_as_default %>
-            <%= link_to subscribe_path_for(subscription, option_params), onclick: '$(this).find("input").prop("checked", true);$(this).parent().parent().parent().next().slideDown();$(this).parent().parent().parent().next().find("input").prop("checked", true);$(this).parent().parent().parent().next().next().slideDown();$(this).parent().parent().parent().next().next().find("input").prop("checked", true);', method: :put, remote: true do %>
+            <%= link_to subscribe_path_for(subscription, option_params.merge(with_email_subscription: ActivityNotification.config.subscribe_to_email_as_default, with_optional_targets: ActivityNotification.config.subscribe_to_optional_targets_as_default)), onclick: "$(this).find(\"input\").prop(\"checked\", true);$(this).parent().parent().parent().next().slideDown();$(this).parent().parent().parent().next().find(\"input\").prop(\"checked\", #{ActivityNotification.config.subscribe_to_email_as_default.to_s});$(this).parent().parent().parent().next().next().slideDown();$(this).parent().parent().parent().next().next().find(\"input\").prop(\"checked\", #{ActivityNotification.config.subscribe_to_optional_targets_as_default});", method: :put, remote: true do %>
               <%= check_box :subscribing, "", { checked: false }, 'true', 'false' %>
               <div class="slider"></div>
             <% end %>
           <% else %>
-            <%= link_to subscribe_path_for(subscription, option_params.merge(with_email_subscription: false)), onclick: '$(this).find("input").prop("checked", true);$(this).parent().parent().parent().next().slideDown();$(this).parent().parent().parent().next().next().slideDown();', method: :put, remote: true do %>
+            <%= link_to subscribe_path_for(subscription, option_params.merge(with_email_subscription: false, with_optional_targets: false)), onclick: '$(this).find("input").prop("checked", true);$(this).parent().parent().parent().next().slideDown();$(this).parent().parent().parent().next().next().slideDown();', method: :put, remote: true do %>
```

Comment:
Same as above

Reply:
Here the original code already contained `with_email_subscription: false`, so I thought we either need to remove that or add optional_targets too. On current master, it looks like this with `config.subscribe_as_default = false` in the initializer:

1. I open the page
![image](https://user-images.githubusercontent.com/1900366/111905538-7a5d1400-8a4c-11eb-9b52-da314112bda7.png)
2. Immediately after I turn on the notifications
![image](https://user-images.githubusercontent.com/1900366/111905570-a082b400-8a4c-11eb-893c-6eaced91a349.png)
3. A few moments later, when the server responded with a partial
![image](https://user-images.githubusercontent.com/1900366/111905607-cc059e80-8a4c-11eb-8d0c-b13065add5ff.png)

With the changes I added to solve your other comment, these `with_*` overrides will become unneccessary, I'm going to remove them in a commit.